src/block/ShulkerBox.php

```diff
@@ -103,4 +103,22 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 
 		return true;
 	}
+
+	public function getCreativeDrops(): array{
+		$shulker = $this->position->getWorld()->getTile($this->position);
+		if($shulker instanceof TileShulkerBox){
+			$empty = true;
+			for($i=0; $i < $shulker->getInventory()->getSize() && $empty; $i++) { 
+				if(!$shulker->getInventory()->getItem($i)->isNull()){
+					$empty = false;
+				}
+			}
+			if(!$empty){
+				$drop = $this->asItem();
+				$this->addDataFromTile($shulker, $drop);
+				return [$drop];
+			}
```

Comment:
You could reduce the code by using this
```suggestion
			if(count($shulker->getInventory()->getContents(false)) > 0){
                $drop = $this->asItem();
                $this->addDataFromTile($shulker, $drop);
                return [$drop];
            }
```

Reply:
* minus the messed-up indentation

Reply:
Indeed, I have made my life more complex for nothing.

Reply:
> * minus the messed-up indentation

I checked twice, why is it messed up again..

Reply:
It should use tabs. GitHub web editor is super messed up.

Reply:
FTFY